server/helpers/createSocialUsers.js

```diff
@@ -0,0 +1,50 @@
+import models from '../database/models';
+import generateToken from './generateToken';
+import dateHelper from './dateHelper';
+
+const { User, Session } = models;
+const { expiryDate } = dateHelper;
+/**
+ * @name createSocialUser
+ * @description function that creates a new user in via socialLogin
+ * @param { Object }  data to check in the database
+ * @returns { Object } value indicating if username exists
+ */
+const createSocialUser = async (data) => {
+  const {
+    firstName,
+    lastName,
+    email,
+    ipAddress,
+    userAgent,
+    devicePlatform
+  } = data;
+  const param = email || firstName;
+  const val = /\D/.test(param) && /@.{2,15}\./.test(param)
+    ? { email: param }
+    : { userName: param };
+  const expiresAt = expiryDate(devicePlatform);
+  const users = await User.findOrCreate({
+    where: val,
+    defaults: {
+      firstName,
+      lastName,
+      email,
+      userName: firstName
```

Comment:
Are you saving firstName as userName? This might not be a good idea because the userName column is unique in the database and many people can share the same first name.

Reply:
I have set it to null, thanks